### PR TITLE
Add a test which checks what custom module gets from the agent

### PR DESCRIPTION
**Pull request by Lex-2008**
Ticket: CFE-3723

### CODE REVIEW

**Comment by Lex-2008** [comment]
@cf-bottom build plz

**Comment by Lex-2008** [comment]
@cf-bottom buid exotics

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7503/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7503/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7503/

**Comment by Lex-2008** [comment]
latest build is green enough: https://ci.cfengine.com/job/pr-pipeline/7516/

File: tests/acceptance/30_custom_promise_types/22_what_module_gets.cf [hunk 1]
```diff
@@ -0,0 +1,57 @@
+######################################################
+#
+#  ...
```
Comment (vpodzime) [inline tests/acceptance/30_custom_promise_types/22_what_module_gets.cf:3]: Some work needed here ;)

File: tests/acceptance/30_custom_promise_types/22_what_module_gets.cf [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+{
+    interpreter => "/bin/bash";
+    path => "$(this.promise_dirname)/cat_module.sh";
+}
+
+bundle agent test
+{
+  meta:
+    "description" -> { "CFE-3723" }
+      string => "...";
```
Comment (vpodzime) [inline tests/acceptance/30_custom_promise_types/22_what_module_gets.cf:40]: And here.

File: tests/acceptance/30_custom_promise_types/22_what_module_gets.cf [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+#######################################################
+
+bundle agent check
+{
+  methods:
+      "any" usebundle => dcs_check_diff("/tmp/module.log",
+                                            "$(this.promise_filename).expected",
+                                            "$(this.promise_filename)");
```
Comment (vpodzime) [inline tests/acceptance/30_custom_promise_types/22_what_module_gets.cf:55]: bad indentation

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise. As long as it works. :)

**Comment by Lex-2008** [comment]
@cf-bottom ask jenkins to check that my recent changes didn't break exotics, plz

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7590/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7590/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7590/

**Review by vpodzime — APPROVED** [review]
Looks promising. Please make sure it really does work on all exotics.

**Comment by Lex-2008** [comment]
Green on PACKAGES_ppc64_aix_53 and PACKAGES_sparc64_solaris_10 
Yellow on PACKAGES_ppc64_aix_71 and PACKAGES_sparc64_solaris_11 (no ssh rute to host)
Let's merge and assume that if it's green on lower versions - it will be green on higher, when they're online

**Pull request merged**